tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,80 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+# note: please be careful if you want to change any of the parameters used in this test.
+# changing parameters without extensive testing may easily lead to weak testing case assertion
+# which means eviction may not get triggered.
+@pytest.mark.asyncio
+async def test_policy_based_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(
+        proactor_threads=1, cache_mode="true", maxmemory="256mb", enable_heartbeat_eviction="false"
+    )
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    pipe = c_master.pipeline(transaction=False)
+    batch_fill_data(client=pipe, gen=gen_test_data(3700000))
```

Comment:
@adiholden and now we also switched to use batch_fill_data... we can now totally avoid use of seeder now... which causes lots of correctness issue

Reply:
this is great. also if you add to this test "debug POPULATE 6000 key 44000" before replication you will be able to get to eviction much much faster the test can pass in few seconds